workers/data_refinery_workers/downloaders/geo.py

```diff
@@ -0,0 +1,374 @@
+                actual_file.has_raw = True
+                actual_file.save()
+
+                original_file_sample_association = OriginalFileSampleAssociation()
+                original_file_sample_association.sample = sample
+                original_file_sample_association.original_file = actual_file
+                original_file_sample_association.save()
+
+                unpacked_sample_files.append(actual_file)
+
+            else:
+                print(og_file['filename'])
+
+    # These files are only gzipped.
+    # These are generally the raw-raw data
+    elif '.gz' in dl_file_path:
+        extracted_files = _extract_gz(dl_file_path, accession_code)
+        unpacked_sample_files = []
+        for og_file in extracted_files:
+
+            filename = og_file['filename']
+            sample_id = filename.split('.')[0]
+
+            try:
+                # The archive we downloaded
+                archive_file = OriginalFile.objects.get(source_filename__contains=filename)
+                archive_file.is_downloaded=True
+                archive_file.is_archive=True
+                archive_file.absolute_file_path = dl_file_path
+                archive_file.calculate_size()
+                archive_file.calculate_sha1()
+                archive_file.save()
+
+                actual_file = OriginalFile()
+                actual_file.is_downloaded=True
+                actual_file.is_archive=False
+                actual_file.absolute_file_path = og_file['absolute_path']
+                actual_file.filename = og_file['filename']
+                actual_file.calculate_size()
+                actual_file.calculate_sha1()
+                actual_file.has_raw = True
+                actual_file.save()
+
+                for sample in related_samples:
+                    new_association = OriginalFileSampleAssociation()
+                    new_association.original_file = actual_file
+                    new_association.sample = sample
+                    new_association.save()
+
+                # Question - do we want to delete this extracted archive file?
+                # archive_file.delete()
+
+                unpacked_sample_files.append(actual_file)
+            except Exception as e:
+                print(e)
+                logger.warn("Found a file we didn't have an OriginalFile for! Why did this happen?: " + og_file['filename'])
+
+    # This is probably just a .txt file
+    else:
+        unpacked_sample_files = []
+
+        filename = dl_file_path.split('/')[-1]
+        sample_id = filename.split('_')[0]
+
+        try:
+            actual_file = OriginalFile()
+            actual_file.is_downloaded=True
+            actual_file.is_archive=False
+            actual_file.absolute_file_path = dl_file_path
+            actual_file.filename = filename
+            actual_file.calculate_size()
+            actual_file.calculate_sha1()
+            actual_file.has_raw = True
+            actual_file.save()
```

Comment:
`is_public` is this something that _we_ are sharing with the public (it's the mildest form of access control) - it's not appropriate for an OriginalFile, since they're never public from us, and are hosted elsewhere (source_url)

Reply:
Ah ok cool. That's good to know.